docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,174 @@
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Linear Algebra
+--------------
+
+Currently, there is very minimal linear algebra operation support for complex tensors.
+We currently support :func:`torch.mv`, :func:`torch.svd`, :func:`torch.qr`, and :func:`torch.inverse`
+(the latter three are only supported on CPU). However we are working to add support for more
+functions soon: :func:`torch.matmul`, :func:`torch.solve`, :func:`torch.eig`, :func:`torch.eig`,
+:func:`torch.symeig`. If any of these would help your use case, please
+`search <https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+complex>`_
+if an issue has already been filed and if not, `file one <https://github.com/pytorch/pytorch/issues/new/choose>`_.
+
+
+Serialization
+-------------
+
+Complex Tensors can be serialized, allowing data to be saved as complex values.
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+
+Autograd
+--------
+
+PyTorch supports Autograd for Complex Tensors. The autograd APIs can be
+used for both holomorphic and non-holomorphic functions. For non-holomorphic
+functions, the gradient is evaluated as if it were holomorphic. For more details,
+check out the note :ref:`complex_autograd-doc`.
+
+Gradient calculation can also be easily done for functions not supported for complex tensors
+yet by enclosing the unsupported operations between :func:`torch.view_as_real` and
+:func:`torch.view_as_complex` functions. The example shown below computes the pointwise multiplication
+of two complex tensors, in one case by performing operations on complex tensors, and in the other
+by by performing operations on complex tensors viewed as real tensors. As shown below, the gradients
+computed have same values in both cases.
+
+::
+
+     >>> x = torch.randn(2, dtype=torch.cfloat, requires_grad=True)
+     >>> y = torch.randn(2, dtype=torch.cfloat, requires_grad=True)
+     >>> x_ = x.detach().requires_grsad_(True)
+     >>> y_ = y.detach().requires_grad_(True)
+     >>> z = x[0]*y[0] + x[1]*y[1]
+     >>> z
+     tensor(0.2114-1.1952j, grad_fn=<AddBackward0>)
+
+     >>> x0 = x_.clone()
+     >>> y0 = y_.clone()
+     >>> x1 = torch.view_as_real(x0)
+     >>> y1 = torch.view_as_real(y0)
+     >>> z_ = torch.empty_like(x1)
+     >>> z_[:, 0] = x1[:, 0] * y1[:, 0] - x1[:, 1] * y1[:, 1]
+     >>> z_[:, 1] = x1[:, 0] * y1[:, 1] + x1[:, 1] * y1[:, 0]
+     >>> z1 = torch.view_as_complex(z_)
+     >>> z2 = z1.sum()
+     >>> z2
+     tensor(0.2114-1.1952j, grad_fn=<SumBackward0>)
+
+     >>> z.backward()
+     >>> z2.backward()
+     >>> x.grad, y.grad
+     tensor([-0.6815+0.5931j,  0.5333-1.0872j]) tensor([-0.4869+0.9011j,  0.3673+0.2007j])
+     >>> x_.grad, y_.grad
+     tensortensor([-0.6815+0.5931j,  0.5333-1.0872j]) tensor([-0.4869+0.9011j,  0.3673+0.2007j])
```

Comment:
looks like a copy paste bug tensortensor

Reply:
fixed! (also updated the example to make it more readable)